src/containers/BooksForm.js

```diff
@@ -10,31 +14,77 @@ const category = [
   { id: 7, name: 'Sci-fi' },
 ];
 
-const BooksForm = () => (
-  <div>
-    <form>
-      <label htmlFor="title">
-        The Title of the Book
-        <input
-          type="text"
-          placeholder="enter the title of the book"
-          name="title"
-          id="title"
-        />
-      </label>
-      <label htmlFor="select">
-        <select>
-          {category.map(c => (
-            <option key={c.id}>{c.name}</option>
-          ))}
-        </select>
-      </label>
-      <button type="submit" value="submit">
-        Submit
-      </button>
-    </form>
-  </div>
-);
-
-
-export default BooksForm;
+class BooksForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      // id: 0,
+      title: '',
+      category: '',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange = (ev) => {
+    ev.preventDefault();
+    const { value, name } = ev.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit = (ev) => {
+    ev.preventDefault();
+    const { title, category } = this.state;
+    const id = Math.floor(Math.random() * 100);
+    const { createBook } = this.props;
+    createBook({ id, title, category });
+
+    this.setState({
+      title: '',
+      category: '',
+    });
+  }
+
+  render() {
+    const { title, category: cat } = this.state;
+    return (
+      <div>
+        <form>
+          <label htmlFor="title">
+            The Title of the Book
+            <input
+              type="text"
+              placeholder="enter the title of the book"
+              value={title}
+              name="title"
+              id="title"
+              onChange={this.handleChange}
+            />
+          </label>
+          <label htmlFor="select">
+            <select name="category" value={cat} onChange={this.handleChange}>
```

Comment:
Please what does the value attribute in this select tag do? I do not think it is needed but if you see a reason why it is please leave a comment